tiledb/sm/c_api/tiledb.h

```diff
@@ -1211,7 +1212,8 @@ TILEDB_EXPORT int tiledb_domain_dump(
  * @param name The dimension name.
  * @param type The dimension type.
  * @param dim_domain The dimension domain.
- * @param tile_extent The dimension tile extent.
+ * @param tile_extent The dimension tile extent. May be `NULL`, in which case
```

Comment:
Hmm, I think that this is valid only for sparse arrays. For dense arrays you will get an error upon schema initialization after you add the dimension to the domain, set the domain to the array schema, and then attempt to create an array with that schema.

Also for sparse arrays, `null` tile extents do not mean "entire domain". They mean no tiling at all (some operations are boosted internally in this case, e.g., comparing the order of two coordinate tuples).